mongo/datadog_checks/mongo/collectors/replica.py

```diff
@@ -0,0 +1,104 @@
+import time
+
+from six.moves.urllib.parse import urlsplit
+
+from datadog_checks.mongo.collectors.base import MongoCollector
+from datadog_checks.mongo.common import REPLSET_MEMBER_STATES, SOURCE_TYPE_NAME, get_state_name
+
+try:
+    import datadog_agent
+except ImportError:
+    from datadog_checks.base.stubs import datadog_agent
+
+
+class ReplicaCollector(MongoCollector):
+    """Collect replica set metrics by running the replSetGetStatus command. Also keep track of the previous node state
+    in order to submit events on any status change.
+    """
+
+    def __init__(self, check, tags, last_state):
+        super(ReplicaCollector, self).__init__(check, "admin", tags)
+        # Members' last replica set states
+        self._last_state = last_state
+
+        # Makes a reasonable hostname for a replset membership event to mention.
+        uri = urlsplit(self.check.clean_server_name)
+        if '@' in uri.netloc:
+            self.hostname = uri.netloc.split('@')[1].split(':')[0]
+        else:
+            self.hostname = uri.netloc.split(':')[0]
+        if self.hostname == 'localhost':
+            self.hostname = datadog_agent.get_hostname()
+
+    def _report_replica_set_state(self, state, replset_name):
+        """
+        Report the member's replica set state
+        * Submit a service check.
+        * Create an event on state change.
+        """
+        if self._last_state == state or self._last_state == -1:
+            return
+
+        status = (
+            REPLSET_MEMBER_STATES[state][1]
+            if state in REPLSET_MEMBER_STATES
+            else 'Replset state %d is unknown to the Datadog agent' % state
+        )
+        short_status = get_state_name(state)
+        last_short_status = get_state_name(self._last_state)
```

Comment:
(Nit) We're mixing `state` and `status` here, we should keep `state` throughout (esp since there's a "status phrase" defined just above). Also I think `short_last_state` ends up easier to read?

```suggestion
        short_state = get_state_name(state)
        short_last_state = get_state_name(self._last_state)
```

Reply:
Using state_id and state_str instead